### PR TITLE
feat: abandoned

### LINKED ISSUE #18784: Append " - abandoned" to PRs which can't be autoclosed
### What would you like Renovate to be able to do?

Indicate in the PR title when a PR has been abandoned.

### If you have any ideas on how this should be implemented, please tell us here.

Today if we intend to autoclose a PR but discover it has been modified then we don't close it, don't modified the title, but we ensure a comment.

In addition, we should modify the PR title with a " - abandoned" suffix similar to " - autoclosed".

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Context
- Closes: #18784 

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
We still want to add the comment like before

**Comment by rarkins** [comment]
Sorry, updated the issue to match

File: lib/workers/repository/finalise/prune.ts [hunk 1]
```diff
@@ -33,8 +33,17 @@ async function cleanUpBranches(
...
           );
           if (GlobalConfig.get('dryRun')) {
-            logger.info(`DRY-RUN: Would add Autoclosing Skipped comment to PR`);
+            logger.info(`DRY-RUN: Would update PR title and ensure comment.`);
           } else {
+            let newPrTitle = pr.title;
+            if (!pr.title.endsWith('- abandoned')) {
+              newPrTitle += ' - abandoned';
+            }
+            await platform.updatePr({
```
Comment (rarkins) [inline lib/workers/repository/finalise/prune.ts:42]: Move this inside the above if

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/finalise/prune.ts [hunk 2]
```diff
@@ -33,8 +33,18 @@ async function cleanUpBranches(
             'Branch is modified - skipping PR autoclosing'
           );
           if (GlobalConfig.get('dryRun')) {
-            logger.info(`DRY-RUN: Would add Autoclosing Skipped comment to PR`);
+            logger.info(`DRY-RUN: Would update PR title and ensure comment.`);
           } else {
+            let newPrTitle = pr.title;
```
Comment (rarkins) [inline lib/workers/repository/finalise/prune.ts:38]: No need for this let now

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by viceice — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**